src/test_l3_dsHdmiIn.c

```diff
@@ -0,0 +1,1461 @@
+                         "selected edid type: %s, edid version:%d\t",
+                 ut_control_GetMapString(dsHdmiInPort_mapTable, port), port,
+                 ut_control_GetMapString(tv_hdmi_edid_version_mapTable, edidver) , edidver);   
+    }
+
+exit:
+UT_LOG_INFO("Out %s", __FUNCTION__);
+}
+
+/**
+* @brief This test to get edidversion.
+*
+* This test function is to get edidversion of HdmiInput port on platform.
+*
+* **Test Group ID:** 03@n
+* **Test Case ID:** 012@n
+*
+* **Test Procedure:**
+* Refer to Test specification documentation
+* [ds-hdmi-in_halSpec.md](../../docs/pages/ds-hdmi-in_halSpec.md)
+*/
+void test_l3_HdmiIn_get_edidversion(void)
+{
+    gTestID = 12;
+    UT_LOG("\n In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
+
+    dsError_t ret   = dsERR_NONE;
+    int select;
+    dsHdmiInPort_t port = dsHDMI_IN_PORT_NONE;
+    tv_hdmi_edid_version_t edidver = HDMI_EDID_VER_MAX;
+    uint8_t numInputPorts = 0;
+
+    numInputPorts = UT_KVP_PROFILE_GET_UINT8("dsHdmiIn/numberOfPorts");
+    printf("\n**********Please select port***********\n");
+    scanf("%d", &select);
+    readAndDiscardRestOfLine(stdin);
+
+    if(select <= 0 || select >numInputPorts)
+    {
+      printf("\n invalid port selected\n");
+      goto exit;
+    }
+    port = select;
+   
+    ret = dsGetEdidVersion(port, &edidver);
+        if(ret != dsERR_NONE)
+    {
+       UT_LOG_ERROR("Result: Failed dsGetEdidVersion Error:[%s]",ut_control_GetMapString(dsError_mapTable, ret));
+    } else {
+           UT_LOG_INFO("Result : Passed dsGetEdidVersion port type: %s, portnumber: %s\t"
+                         "retrieved edid type: %s, edid version:%d",
+                 ut_control_GetMapString(dsHdmiInPort_mapTable, port), port,
+                 ut_control_GetMapString(tv_hdmi_edid_version_mapTable, edidver) , edidver);
+    }
+exit:
+UT_LOG_INFO("Out %s", __FUNCTION__);
+}
+
+/**
+* @brief This test to get allmstatus.
+*
+* This test function is to get allmstatus of HdmiInput port on platform.
+*
+* **Test Group ID:** 03@n
+* **Test Case ID:** 013@n
+*
+* **Test Procedure:**
+* Refer to Test specification documentation
+* [ds-hdmi-in_halSpec.md](../../docs/pages/ds-hdmi-in_halSpec.md)
+*/
+void test_l3_HdmiIn_get_allmstatus(void)
+{
+    gTestID = 13;
+    UT_LOG("\n In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
+
+    dsError_t ret   = dsERR_NONE;
+    int select = 0;
+    dsHdmiInPort_t port = dsHDMI_IN_PORT_NONE;
+    bool allmstatus = false;
+    uint8_t numInputPorts = 0;
+
+    numInputPorts = UT_KVP_PROFILE_GET_UINT8("dsHdmiIn/numberOfPorts");
+    printf("\n**********Please select port***********\n");
+    scanf("%d", &select);
+    readAndDiscardRestOfLine(stdin);
+
+    if(select <= 0 || select >numInputPorts)
+    {
+      printf("\n invalid port selected\n");
+      goto exit;
+    }
+    port = (dsHdmiInPort_t)select;
+
+    ret = dsGetAllmStatus(port , &allmstatus);
+    if(ret != dsERR_NONE)
```

Comment:
Remove the else case

In all cases, you need to see the results output

You've got some serious debugging issues todo if this was to assert and a failure of L1-L2 to fix in that case.

Reply:
done